### PR TITLE
Feature/setup continued

### PR DESCRIPTION
**Pull request by joemewes**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,5 @@ yarn-debug.log*
 yarn-error.log*
 /.htpasswd
 /.scannerwork
+/dist
+/.env
```
Comment (jamiemagique) [inline .gitignore]: 👍

File: package.json [hunk 2]
```diff
@@ -4,34 +4,57 @@
   "description": "",
   "main": "index.js",
   "scripts": {
+    "precommit": "lint-staged",
     "test": "./node_modules/.bin/jest --coverage",
-    "start":
-      "webpack-dev-server --open --config ./webpack.dev.js --mode development",
-    "build:server": "node server",
-    "start:server": "./node_modules/.bin/nodemon ./server/index.js 9000",
+    "start:server": "babel-node server",
```
Comment (jamiemagique) [inline package.json]: You didn't like my renaming to make the script names consistent with one another?

File: serve.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import express from 'express';
+import morgan from 'morgan';
+import path from 'path';
+import webpack from 'webpack';
+import webpackMiddleware from 'webpack-dev-middleware';
+import webpackConfig from './config/webpack.config.dev.js';
+
+require('dotenv').config();
```
Comment (jamiemagique) [inline serve.js]: Interesting: https://github.com/motdotla/dotenv/tree/master/examples/es6

File: config/webpack.config.common.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    extensions: ['*', '.js', '.jsx'],
+  },
+  plugins: [
+    new CleanWebpackPlugin(['dist']),
+    new HtmlWebpackPlugin({
+      title: 'Production',
+      template: './client/public/index.html',
+    }),
+    new webpack.DefinePlugin({
+      'process.env.BROWSER': true,
```
Comment (jamiemagique) [inline config/webpack.config.common.js]: Need to add this to the readme, along with the others.

File: package.json (hunk 2 above)
Comment (joemewes) [reply package.json]: my choice was arbitrary... I've changed this file so much in the last 24 hours I didn't know what was mine, yours ro rubbish.  MORE than happy to revert


what should this be? `build:server` ??

File: serve.js (hunk 3 above)
Comment (joemewes) [reply serve.js]: but in `../../lib/main` they're importing via require()...

To get the `build` to work via babel from webpack, I had to do this https://github.com/4AllDigital/react-express-babel-webpack4/blob/feature/setup-continued/webpack.config.js

File: package.json (hunk 2 above)
Comment (jamiemagique) [reply package.json]: I was trying to group them by their environment, so:

```
start: dev start
start:server: dev server start
build: production build
build:server: production server build
...
```

Comment (jamiemagique) [reply package.json]: You _could_ have:

```
start: npm run start:client && npm run start:server
...
```

But you don't get the nice logs.

Comment (joemewes) [reply package.json]: Think I've put it back  in order and names

Comment (joemewes) [reply package.json]: we have this in CA... but TBH I never use : `"dev": "npm-run-all --parallel dev:server start",`

**Pull request merged**